java/ql/src/Security/CWE/CWE-524/SensitiveKeyboardCache.qhelp

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>When a user enters information to a text input field on an Android application, then by default their input is saved to a keyboard cache, 
+which provides autocomplete suggestions and predictions. If the input field is expected to contain sensitive information, such as a password or banking details, 
+this sensitive data may be leaked to other applications via the keyboard cache.</p>
+
+</overview>
+<recommendation>
+
+<p>For input fields expected to accept sensitive information, an input type such as <code>"textNoSuggestions"</code> (or <code>"textPassword"</code> for a password) 
+should be used to ensure that the input does not get stored in the keyboard cache.</p>
+<p>The input type can also be set in code through <code>TextView.setInputType()</code> rather than declared through XML.</p>
```

Comment:
```suggestion
<p>In addition to declaring an input type through XML, you can also set the input type through code using <code>TextView.setInputType()</code>.</p>
```

Reply:
How accurately does this convey that setting the input type in code is an alternative to setting it in XML? To me this wording suggests that it's something you'd do as well as declare it in XML, which isn't necassary.

Reply:
Good point, I've updated my suggestion above to make that more clear 👍🏼